cadence/scripts/ListUserDappies.cdc

```diff
@@ -1,9 +1,70 @@
 import DappyContract from "../contracts/DappyContract.cdc"
+import MetadataViews from "../contracts/MetadataViews.cdc";
 
-pub fun main(addr: Address): {UInt64: DappyContract.Template} {
+pub struct DappyDetails {
```

Comment:
It has no use in the contract as the data structure but can define in the contract as well, But I think it will just eat space besides making the system better,

Reply:
It doesn't have use in the contract necessarily, but it might be more useful to have it centrally defined instead of having to define it in every script. For cryptodappy, it doesn't really matter because it isn't a production level contract that will actually be in use in the real world, but might be a good practice in smart contracts in general. But if you don't think it should be included in the contract, then I'll defer to you.

Reply:
Yeah, I agree with you that it is good to have it in the smart contract for the production-grade system so everyone can use that data structure.